### PR TITLE
GH-43870: [C++][Acero] Fix typos in join benchmark

### PR DESCRIPTION
**Pull request by zanmato1984**
### Rationale for this change

These are rather obvious typos.

### Are there any user-facing changes?

* GitHub Issue: #43870

### CODE REVIEW

**Comment by zanmato1984** [comment]
Trivial and quick fix. @pitrou @felipecrv @mapleFU Could you help to merge? Thanks.

**Review by mapleFU — APPROVED** [review]

**Comment by mapleFU** [comment]
Just curious how does the benchmark like before this fix?

**Review by kou — APPROVED** [review]
+1

**Comment by zanmato1984** [comment]
> Just curious how does the benchmark like before this fix?

Sorry I didn't check that. But either better or worse, the baseline has to be reset IMO.

File: cpp/src/arrow/acero/hash_join_benchmark.cc [hunk 1]
```diff
@@ -104,7 +104,7 @@ class JoinBenchmark {
       key_cmp.push_back(JoinKeyCmp::EQ);
     }
 
-    for (size_t i = 0; i < settings.build_payload_types.size(); i++) {
+    for (size_t i = 0; i < settings.probe_payload_types.size(); i++) {
```
Comment (pitrou) [inline cpp/src/arrow/acero/hash_join_benchmark.cc:107]: Does this change the benchmark results somehow?

Comment (zanmato1984) [reply cpp/src/arrow/acero/hash_join_benchmark.cc:107]: It is supposed to. But I didn't review the complete benchmarks that are utilizing this piece of code. There might be a chance that all benchmarks are using the same number of payload columns thus no result change. 

Is there anything we should be preparing for a potential benchmark result change?

Comment (pitrou) [reply cpp/src/arrow/acero/hash_join_benchmark.cc:107]: Well, it seems the "detailed" benchmarks all have:
```c++
  settings.build_payload_types = {binary()};
  settings.probe_payload_types = {binary()};
```
while the "basic" benchmarks leave those settings empty? (is it desired?)

Comment (pitrou) [reply cpp/src/arrow/acero/hash_join_benchmark.cc:107]: Ah, it would actually matter for `BM_HashJoinBasic_PayloadSize`.

Comment (pitrou) [reply cpp/src/arrow/acero/hash_join_benchmark.cc:107]: Ok, it seems to require OpenMP, so I won't try running it.

**Review by pitrou — APPROVED** [review]

**Pull request merged**